File: app-presentation/src/main/java/com/zxjdev/smile/presentation/infrastucture/main/MainContract.java
package com.zxjdev.smile.presentation.infrastucture.main;

import com.zxjdev.smile.presentation.common.base.BasePresenter;
import com.zxjdev.smile.presentation.common.base.BaseView;
import com.zxjdev.smile.presentation.user.UserModel;

public interface MainContract {

  interface View extends BaseView {

    void displayUser(UserModel user);
  }

  interface Presenter extends BasePresenter<View> {

    void handleChangeAvatar(String picturePath);
  }
}

File: app-presentation/src/main/java/com/zxjdev/smile/presentation/infrastucture/main/MainContract.kt
package com.zxjdev.smile.presentation.infrastucture.main

import com.zxjdev.smile.presentation.common.base.BasePresenter
import com.zxjdev.smile.presentation.common.base.BaseView
import com.zxjdev.smile.presentation.user.UserModel

interface MainContract {

    interface View : BaseView {

        fun displayUser(user: UserModel)
    }

    interface Presenter : BasePresenter<View> {

        fun handleChangeAvatar(picturePath: String?)
    }
}
